deploy/tests/test_utils.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+from unittest.mock import MagicMock, call
+from util.fsspec_utils import downloadURI
+from fsspec.implementations.local import LocalFileSystem
+
+
+def test_download_uri(mocker):
+    logger_mock = mocker.patch("util.fsspec_utils.rootLogger", MagicMock())
+    local_test_path = Path("tests/spec_results_download.csv")
+    test_uri = "https://www.spec.org/cpu2017/results/res2018q1/cpu2017-20171224-02028.csv"
```

Comment:
I don't want the tests to actually go out onto the internet to fetch resources from machines we have no control over while testing.  Do the fsspec tests themselves actually go out on the network or do they mock the network stuff out?

We should be able to use `moto` and and S3 URI and not require the actual internet to be working for the tests to pass.

Reply:
Refer to https://github.com/filipstamenkovic-sifive/firesim/blob/fsspec_for_bitstream/deploy/tests/awstools/test_awstools.py for examples of using moto to mock the ec2 API.

Reply:
Yeah that makes sense, I could see internet not working in for example a Github Actions environment blocking a CI pipeline because of this test. I'll change it so we mock S3 instead.

Reply:
Sounds good.  We're not really interested in testing http support but https://github.com/fsspec/filesystem_spec/blob/c99eca4d65eeede94be3f1eb1cd8d324571b821a/fsspec/tests/conftest.py#L146-L149 is the fixture fsspec uses to provide a local http server.

I think we want to have `file`, `ssh` and `s3` tests to start.  https://github.com/fsspec/filesystem_spec/blob/c99eca4d65eeede94be3f1eb1cd8d324571b821a/fsspec/implementations/tests/test_sftp.py#L20-L55 is where fsspec defines it's fixture for a local ssh server that runs inside a docker container (or skips the test if docker is not available).

@davidbiancolin and @abejgonzalez for testing that our integration of fsspec works, what do you think about depending on a nested docker container in CI?  Since we already spin up multiple servers in CI, perhaps we could:
1. write an integration test script at a higher level that knows about the other servers in CI
2. have the docker version in pytests for people who want to run it locally without other servers
3. skip it if docker isn't available.

Reply:
Unfortunately it seems like fsspec uses HTTP to interact with the S3 REST API and not boto3 for it's S3 integration. When I instantiate a file system using `url_to_fs, it's a `fsspec.implementations.http.HTTPFileSystem` instance, so using moto3 won't work to mock calls to AWS.

Reply:
To mock calls to S3, I think that I'll just have to mock HTTP calls out. I'll take a look at how fsspec does it in their tests.

Reply:
Nevermind, I learned how to properly generate an S3 URI for the test, so fsspec properly instantiates an `S3FileSystem` that wraps boto3... Now I can use moto to mock calls to s3.